src/AuthLDAP.php

```diff
@@ -1917,6 +1917,16 @@ public static function searchForUsers(
                     $sr === false
                     || @ldap_parse_result($ds, $sr, $errcode, $matcheddn, $errmsg, $referrals, $controls) === false
                 ) {
+                    if (ldap_errno($ds) === -1) {
```

Comment:
I do not find any `-1` references in standard error codes nor in php documentation... Where does that comes from?

Reply:
To me; the `ldap_error($ds)` information should be logged on each LDAP error, trying to give us more information on what goes wrong.

About returning `false` or calling`exit`... Well, does that work well from cli aswell as web ui?

Reply:
> It's the value that comes up in my test logs and in those of the customer who reported the case.

It's important to be sure that error code is the same for "all" servers we can work with (at least openldap and active directory - I have no idea if there are other possible platforms).

Reply:
It would be better to throw an exception instead of exiting the script.

Reply:
I use Openldap and the client uses Active Directory, so it's obviously -1 in both cases.

The last commit throws an exception when the error code is not 32.

Reply:
This seems too complex change for 10.0/bf; this kind of change should target main